stdlib/mexpr/cfa.mc

```diff
@@ -1371,9 +1371,9 @@ end
 -- probably be added.
 lang RefOpCFA = CFA + ConstCFA + RefOpAst
   sem generateConstraintsConst info ident =
-  -- | CRef _ -> []
+  | CRef _ -> []
+  | CDeRef _ -> []
```

Comment:
This is not correct in the general case, now we do not track abstract values (i.e., abstract lambdas and constants) flowing through references. We need to do something similar as in the sequence operations (but much simpler as references only contain one item). By the way, what is the purpose of having references without `modref`? I recall we discussed it, but I have forgotten.

Reply:
I see. I'm storing a continuously updated timestamp, which I compute other relative timestamps to, and this can also take place inside a model. However, I know I will never use a `modref` inside a model as the time updates always take place outside of the model code. It's something along the lines of
```
let currTime = ref (getWallTime ()) in
...
let model = lam.
  let absTime = getWallTime () in
  let relTime = timeDiff absTime (deref currTime) in
  ...
in
...
infer ... (model ())
```
Because of the lambda lifting, I believe the `currTime` reference will be captured as a parameter to `model`. I could thread along the timestamp(s) manually (without using references) to all functions, but that's a lot of extra work, so I did it this way mainly to save time.

Reply:
Ah, right, now I remember. I can have a look tomorrow at implementing a correct implementation of `ref`/`deref` in the CFA framework.